### PR TITLE
Feature SVG-use Support #7410

### PR DESCRIPTION
**Pull request by pahy**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by hawkeye64** [comment]
@pahy Nice! I like it! :heart:

File: ui/src/components/icon/QIcon.js [hunk 1]
```diff
@@ -167,6 +175,16 @@ export default Vue.extend({
       return h('svg', data, mergeSlot(this.type.nodes, this, 'default'))
     }
 
+    if (this.type.svguse === true) {
+      data.attrs.focusable = 'false' /* needed for IE11 */
+      data.attrs.viewBox = this.type.viewBox
+
+      const child = [
+        h('use', { attrs: { 'xlink:href' : this.type.src } }, mergeSlot(this.type.nodes,this, 'default'))
```
Comment (hawkeye64) [inline ui/src/components/icon/QIcon.js:183]: Could probably add a space before `,this`, other than that this looks good.

File: ui/src/components/icon/QIcon.js [hunk 2]
```diff
@@ -167,6 +175,16 @@ export default Vue.extend({
...
+      ]
+      return h('svg', data, child)
```
Comment (webnoob) [inline ui/src/components/icon/QIcon.js:185]: I'm slightly torn on this one but I think I'd prefer to see it written like so to make things more readable (obviously replacing the `child` const.

```
return h('svg', data, [
  h('use', {
    attrs: {
      'xlink:href': this.type.src
    }
  }),
  mergeSlot(this.type.nodes,this, 'default')
])
```

**Comment by webnoob** [comment]
I've added a comment about a change in the return. I think it'll be more readable that way.

Great feature, well done. I'll merge following the requested change (my requested change will also resolve @hawkeye64 )

**Comment by webnoob** [comment]
You've broken the return in your first commit. You've removed the main tag return and there is still a missing space between the `this` param :)

**Comment by pahy** [comment]
What a bummer! One (me) should always do it cleverly and not quickly in between.

**Review by pdanpdan — APPROVED** [review]
LGTM

**Comment by webnoob** [comment]
@pahy It happens :) Thanks for your time on this.

Keep the PR's coming! ;)

File: ui/src/components/icon/QIcon.js (hunk 2 above)
Comment (hawkeye64) [reply ui/src/components/icon/QIcon.js]: I totally agree with this. Not sure how I missed this one.

**Pull request merged**